src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -95,95 +109,41 @@ public override void Initialize(AnalysisContext analysisContext)
                 return;
             }
 
-            var allInterfacesStatus = default(CollectionsInterfaceStatus);
-            foreach (var @interface in namedTypeSymbol.AllInterfaces)
+            var allInterfaces = ImmutableHashSet.CreateRange(namedTypeSymbol.AllInterfaces.Select(i => i.OriginalDefinition));
```

Comment:
```suggestion
            using var allInterfaces = PooledHashSet.GetInstance(namedTypeSymbol.AllInterfaces.Select(i => i.OriginalDefinition));
```

Reply:
Suggested overload doesn't exist but I will add it.